raft/raftpb/raft.proto

```diff
@@ -75,9 +76,26 @@ message HardState {
 	optional uint64 commit = 3 [(gogoproto.nullable) = false];
 }
 
+// ConfChangeTransition specifies the behavior of a configuration change with
+// respect to joint consensus.
+enum ConfChangeTransition {
+	// Automatically use the simple protocol if possible, otherwise fall back
+	// to ConfChangeJointImplicit.
+	ConfChangeTransitionAuto          = 0;
+	// Use joint consensus, but transition out of the joint configuration
+	// automatically by proposing a no-op configuration change.
+	ConfChangeTransitionJointImplicit = 1;
+	// Use joint consensus and remain in the joint configuration until the
+	// application proposes a no-op configuration change.
+	ConfChangeTransitionJointExplicit = 2;
```

Comment:
I added some more color to the comments. I also think this will be useful for testing (I'm working on a datadriven framework that gives you essentially a "raft repl"), though this alone wouldn't be enough justification.

We'll use auto in CRDB (and have some custom code to synthesize the raft command we need with the trigger when we see the trigger), but if we wanted to sync the descriptor with the joint config, it would be easier if we could transition out manually. Etcd tends to be less flexible with on-the-fly stuff (see UpdateNode) so if they start using joint configs they may prefer the explicit step over refactoring a bunch of their code.